systers_portal/users/adapter.py

```diff
@@ -1,9 +1,61 @@
 from allauth.account.adapter import DefaultAccountAdapter
 from django.core.urlresolvers import reverse
+from django.core.exceptions import ValidationError
+import re
 
 
 class SystersUserAccountAdapter(DefaultAccountAdapter):
     """Custom account adapter with different than default redirect URLs"""
+
+    def clean_username(self, username, shallow=False):
+        if len(username) < 3:
+            raise ValidationError("Username must be atleast 3 characters long")
+
+    def clean_password(self, password):
+        # Password should have at least one uppercase letter , one digit and one special character
+        x = r'^(?=.*?[0-9])(?=.*[A-Z])'
+        y = '[~!@#$%^&*()_+{}":;\']+$'
+        z = r'^(?=.*?[0-9])'
+        u = r'^(?=.*?[A-Z])'
+        a = re.match(z, password)
+        b = re.match(x, password)
+        c = set(y).intersection(password)
+        d = re.match(u, password)
```

Comment:
[There is a need to rename these variables](https://www.python.org/dev/peps/pep-0008/#id37).